### PR TITLE
Clean up unused styling (CSS/SCSS)

### PR DESCRIPTION
**Pull request by bartkamphorst**
There is still quite a lot of SCSS in gollum that is no longer used, or at least no longer needed, since introducing Primer. This PR attempts some cleaning up.

It also upgrades Primer CSS manually from v.15 to v.20.8.0.

### CODE REVIEW

**Comment by bartkamphorst** [comment]
@benjaminwil Would this endeavour be something you would be willing to collaborate on?

**Comment by benjaminwil** [comment]
@bartkamphorst Sure!

It may make sense add additional pull requests that remove focused chunks of CSS in case we need to revert any of these changes. Do you want me to start my own PR or two?

**Comment by bartkamphorst** [comment]
Great! I was thinking we could both use this branch and double check each other's deletions through manual inspection. But if you think having multiple PRs makes more sense, then that's fine too of course. This is not my area of expertise, so I'm just slowly going through the styling to see what is de facto no longer used and what could be removed if we add a primer class to a certain element, etc.

**Comment by bartkamphorst** [comment]
In any case, it would be great to have another pair of eyes on the deletions I've suggested so far.

**Comment by dometto** [comment]
This would be extremely helpful for improving maintainability! I suppose it probably makes sense to hold off on [moving primer to yarn](https://github.com/gollum/gollum/pull/1914) until this is done.

**Comment by bartkamphorst** [comment]
> This would be extremely helpful for improving maintainability! I suppose it probably makes sense to hold off on https://github.com/gollum/gollum/pull/1914 until this is done.

I don't think that should matter too much, right @benjaminwil ? We will mostly be removing unused lines of SCSS, so as long as primer is available, through local bundling or through yarn, everything should be alright. But perhaps I'm being naive?

**Comment by benjaminwil** [comment]
Yes, I think these two changes shouldn't effect each other. We should be able to continue work on both of them simultaneously.

**Comment by benjaminwil** [comment]
I had a thought re: this pull request, and the one where we upgrade Primer.

We might be able to use a visual diffing tool to take screenshots of every typical path and and compare them against screenshots of `master`. 

I haven't looked into this at all yet, but there's a gem called `capybara-screenshot-diff`, and someone else [wrote an article about setting this up for a Rails app](https://medium.com/@yoooodaaaa/visual-regression-testing-in-capybara-6c147fdceee6).

**Comment by bartkamphorst** [comment]
@benjaminwil @dometto I noticed that we are running gollum with `container-lg` while Github uses `container-xl`. Is this something we want to consider as well for 6.x?
Screenshot 2022-12-24 at 20 53 30

Screenshot 2022-12-24 at 20 55 37

The latter does feel a lot more spacious!

**Comment by benjaminwil** [comment]
If we are voting I would vote `-lg` as I have trouble reading very long lines of text. 🤣 But Gollum isn't just for me, it's for everyone! I will support whatever decision you make.

I haven't had a chance to jump into this CSS myself, yet, sorry!

File: lib/gollum/public/gollum/javascript/editor/gollum.editor.js [hunk 1]
```diff
@@ -1009,7 +1009,7 @@
       }
       var helpData = Help._HELP[name];
 
-      // clear this shiz out
+      // clear out help elements
```
Comment (dometto) [inline lib/gollum/public/gollum/javascript/editor/gollum.editor.js:1012]: :+1: 😆

**Comment by bartkamphorst** [comment]
Okay so I've done quite a bit of cleanup. @benjaminwil @dometto Would you be willing to take this branch for a spin to see if it still works for you as you would expect? 

There's still a few areas that can be improved, but to do that it would be good to first upgrade to a newer version of Primer. So I propose we first assess this PR, then we work on upgrading Primer, and then we remove more superfluous styling.

**Comment by dometto** [comment]
> @benjaminwil @dometto I noticed that we are running gollum with container-lg while Github uses container-xl. Is this something we want to consider as well for 6.x?

I think this is a nice idea, but maybe worth looking at what that looks like on a page with header/sidebars/footer. Similar to @benjaminwil I also think we should do what most users consider nicest, but lacking any way to do a poll I think the best thing is to go with our own taste! We could also open a Discussion and see if anyone chips in, and/or give users a thread to complain if they don't like the change. :)

**Comment by dometto** [comment]
@bartkamphorst I wonder if it makes sense to also prune some unused CSS classes from our templates, as identified [here](https://github.com/gollum/gollum/pull/1914#issuecomment-1362940141), in this PR? If you think it's too much work or otherwise not worth it, of course don't bother!

**Comment by dometto** [comment]
Thanks for the primer upgrade! Curious what happens with the final step to `v20.8`.

File: lib/gollum/templates/editor.mustache [hunk 2]
```diff
@@ -21,28 +21,38 @@
...
-			<button class="btn btn-sm function-button" id="function-hr" title="Horizontal rule">{{#octicon}}horizontal-rule{{/octicon}}</button>
-			<span class="pr-2"></span>
-			<button class="btn btn-sm function-button" id="function-h1" title="Heading 1">h1</button>
-			<button class="btn btn-sm function-button" id="function-h2" title="Heading 2">h2</button>
-			<button class="btn btn-sm function-button" id="function-h3" title="Heading 3">h3</button>
-			<span class="pr-2"></span>
-			<button class="btn btn-sm function-button" id="function-link" title="Link">{{#octicon}}link{{/octicon}}</button>
-			<button class="btn btn-sm function-button" id="function-image" title="Image">{{#octicon}}image{{/octicon}}</button>
-			<span class="pr-2"></span>
+	<div id="gollum-editor-function-bar" class="border-bottom clearfix mb-2">
```
Comment (dometto) [inline lib/gollum/templates/editor.mustache:24]: Indentation?

File: lib/gollum/templates/editor.mustache [hunk 3]
```diff
@@ -21,28 +21,38 @@
...
+    <div id="gollum-editor-function-buttons" class="overflow-auto no-wrap">
+      <div class="BtnGroup">
+        <button class="btn btn-sm BtnGroup-item function-button" id="function-h1" title="Heading 1">h1</button>
```
Comment (dometto) [inline lib/gollum/templates/editor.mustache:27]: Indentation?

**Comment by dometto** [comment]
Took this branch for a spin and all looks good to me. The changes in the SCSS and templates also seem to make sense to me, although it's difficult to do a proper review of this. But the gains in terms of cleanup are great, so I'm all for merging this! If this introduces any small errors in the layout, it will be much easier to fix now.

@bartkamphorst are you planning to attempt the final upgrade to Primer in this branch, too?

File: lib/gollum/templates/editor.mustache (hunk 3 above)
Comment (bartkamphorst) [reply lib/gollum/templates/editor.mustache:27]: Fixed.

File: lib/gollum/templates/editor.mustache (hunk 2 above)
Comment (bartkamphorst) [reply lib/gollum/templates/editor.mustache]: Fixed.

**Comment by bartkamphorst** [comment]
> are you planning to attempt the final upgrade to Primer in this branch, too?

Yes. Surprisingly though, I couldn't reproduce the issues you encountered in #1914 . Have a look for yourself with the latest commits.

**Comment by dometto** [comment]
> Yes. Surprisingly though, I couldn't reproduce the issues you encountered in https://github.com/gollum/gollum/pull/1914 . Have a look for yourself with the latest commits.

Yes, seems like those issues are being caused by attempting to import the latest version of `primer` from `node_modules`! That's good in itself, although it means more puzzling is required to move the dependence to `yarn`.

All green lights for this PR from me!

**Comment by bartkamphorst** [comment]
@benjaminwil Is this PR okay to merge?

**Comment by benjaminwil** [comment]
@bartkamphorst Hey, I apologize for not responding. I will review this PR even though it's already been merged. From my first skim it seems great. 😄

**Comment by bartkamphorst** [comment]
> @bartkamphorst Hey, I apologize for not responding. I will review this PR even though it's already been merged. From my first skim it seems great. 😄

No worries! Any help is appreciated.

**Comment by benjaminwil** [comment]
Reviewed! All of the changes seem reasonable! Haven't noticed any style issues browsing locally yet.

**Pull request merged**